diff --git a/src/com/android/settings/wifi/WifiConfigController.java b/src/com/android/settings/wifi/WifiConfigController.java
index cf2de8a..f345daa 100644
--- a/src/com/android/settings/wifi/WifiConfigController.java
+++ b/src/com/android/settings/wifi/WifiConfigController.java
@@ -71,6 +71,7 @@ import java.net.InetAddress;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Iterator;
+import android.widget.Toast;
 
 /**
  * The class for allowing UIs like {@link WifiDialog} and {@link WifiConfigUiBase} to
@@ -231,7 +232,31 @@ public class WifiConfigController implements TextWatcher,
             /// M: modify to solve CR:ALPS01851640 @{
             if (mPasswordView == null) {
                 mPasswordView = (TextView) mView.findViewById(R.id.password);
-                mPasswordView.addTextChangedListener(this);
+                mPasswordView.addTextChangedListener(new TextWatcher() {
+
+                        @Override
+                        public void onTextChanged(CharSequence s, int start, int before, int count) {
+                        }
+
+                        public void beforeTextChanged(CharSequence s, int start, int count,int after) {
+                        }
+
+                        public void afterTextChanged(Editable s) {
+                             int length = mPasswordView.getText().toString().length();
+                             android.util.Log.i("zhangweigang", "1233length=:"+length+"----s=:"+s);
+                                     if(length >= SSID_MAX_LEN){
+                                         mPasswordView.setFilters(new InputFilter[] { new Utf8ByteLengthFilter(SSID_MAX_LEN) });
+                                         Toast.makeText(mContext, "The maximum length of the password is 32 bits", Toast.LENGTH_SHORT).show();
+                                     }
+
+                             mTextViewChangedHandler.post(new Runnable() {
+                                         public void run() {
+                                             showWarningMessagesIfAppropriate();
+                                             enableSubmitIfAppropriate();
+                                         }
+                                     });
+                        }
+                });
                 ((CheckBox) mView.findViewById(R.id.show_password))
                     .setOnCheckedChangeListener(this);
             }
@@ -807,7 +832,31 @@ public class WifiConfigController implements TextWatcher,
 
         if (mPasswordView == null) {
             mPasswordView = (TextView) mView.findViewById(R.id.password);
-            mPasswordView.addTextChangedListener(this);
+            mPasswordView.addTextChangedListener(new TextWatcher() {
+
+                            @Override
+                            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                            }
+
+                            public void beforeTextChanged(CharSequence s, int start, int count,int after) {
+                            }
+
+                            public void afterTextChanged(Editable s) {
+                                 int length = mPasswordView.getText().toString().length();
+                                 android.util.Log.i("zhangweigang", "1233length=:"+length+"----s=:"+s);
+                                         if(length >= SSID_MAX_LEN){
+                                             mPasswordView.setFilters(new InputFilter[] { new Utf8ByteLengthFilter(SSID_MAX_LEN) });
+                                             Toast.makeText(mContext, "The maximum length of the password is 32 bits", Toast.LENGTH_SHORT).show();
+                                         }
+ 
+                                 mTextViewChangedHandler.post(new Runnable() {
+                                                 public void run() {
+                                                     showWarningMessagesIfAppropriate();
+                                                     enableSubmitIfAppropriate();
+                                                 }
+                                             });
+                            }
+            });
             mPasswordView.setOnEditorActionListener(this);
             mPasswordView.setOnKeyListener(this);
             ((CheckBox) mView.findViewById(R.id.show_password))
@@ -1223,6 +1272,7 @@ public class WifiConfigController implements TextWatcher,
 
     @Override
     public void afterTextChanged(Editable s) {
+
         mTextViewChangedHandler.post(new Runnable() {
                 public void run() {
                     showWarningMessagesIfAppropriate();
