diff --git a/mediatek/proprietary/external/charger/charging_control.cpp b/mediatek/proprietary/external/charger/charging_control.cpp
index 3332df1..8648478 100644
--- a/mediatek/proprietary/external/charger/charging_control.cpp
+++ b/mediatek/proprietary/external/charger/charging_control.cpp
@@ -215,7 +215,7 @@ static int on_uevent(const char *buf, int len_buf)
 	{
 		bc = get_capacity();
 
-		if (bc >= 90) {
+		if (bc >= 100) {
 			lights_full();
 		} else {
 			if (nCurrentState != LIGHTS_STATE_CHGON)
@@ -395,6 +395,11 @@ void charging_control()
 	if (!is_charging_source_available()) {
 		lights_exit();
 	}
+    //add by mtk begin
+    KPOC_LOGI("on_uevent test by MTK.\n");
+    set_light_state(LIGHTS_CHGON);
+    lights_on();
+    //add end
 
 	pthread_mutex_init(&mutexlstate, NULL);
 
