diff --git a/base/services/core/java/com/android/server/am/ProcessList.java b/base/services/core/java/com/android/server/am/ProcessList.java
index 41e7e16..d0ecd86 100644
--- a/base/services/core/java/com/android/server/am/ProcessList.java
+++ b/base/services/core/java/com/android/server/am/ProcessList.java
@@ -144,7 +144,7 @@ final class ProcessList {
     // we have no limit on the number of service, visible, foreground, or other such
     // processes and the number of those processes does not count against the cached
     // process limit.
-    static final int MAX_CACHED_APPS = 32;
+    static final int MAX_CACHED_APPS = 12;
 
     // We allow empty processes to stick around for at most 30 minutes.
     static final long MAX_EMPTY_TIME = 30*60*1000;
