diff --git a/mediatek/proprietary/bootable/bootloader/preloader/custom/nd387b/cust_bldr.mak b/mediatek/proprietary/bootable/bootloader/preloader/custom/nd387b/cust_bldr.mak
index aa4b0d0..58ccc66 100644
--- a/mediatek/proprietary/bootable/bootloader/preloader/custom/nd387b/cust_bldr.mak
+++ b/mediatek/proprietary/bootable/bootloader/preloader/custom/nd387b/cust_bldr.mak
@@ -10,7 +10,7 @@ endif
 CFG_UART_LOG :=UART1
 CFG_UART_META :=UART1
 CFG_USB_UART_SWITCH := 0
-CFG_TEE_SUPPORT = 0
+CFG_TEE_SUPPORT = 1
 CFG_TRUSTONIC_TEE_SUPPORT = 0
-CFG_MICROTRUST_TEE_SUPPORT = 0
+CFG_MICROTRUST_TEE_SUPPORT = 1
 CFG_GOOGLE_TRUSTY_SUPPORT=0
diff --git a/mediatek/proprietary/trustzone/atf/v1.0/plat/mt6735/drivers/devapc/devapc.c b/mediatek/proprietary/trustzone/atf/v1.0/plat/mt6735/drivers/devapc/devapc.c
index 545a947..0cfc6eb 100755
--- a/mediatek/proprietary/trustzone/atf/v1.0/plat/mt6735/drivers/devapc/devapc.c
+++ b/mediatek/proprietary/trustzone/atf/v1.0/plat/mt6735/drivers/devapc/devapc.c
@@ -335,7 +335,7 @@ static DEVICE_INFO D_APC0_Devices_TEE[] = {
     {"I2C0",                                    E_L0,  E_L3 ,  E_L0, E_L0},
     {"I2C1",                                    E_L0,  E_L3 ,  E_L0, E_L0},
     {"I2C2",                                    E_L0,  E_L3 ,  E_L0, E_L0},
-    {"SPI0",                                    E_L0,  E_L0 ,  E_L0, E_L0},
+    {"SPI0",                                    E_L1,  E_L3 ,  E_L0, E_L0},
     {"PTP_THERMAL_CTL",                         E_L0,  E_L0 ,  E_L0, E_L0},
     {"BTIF",                                    E_L0,  E_L3 ,  E_L0, E_L0},
     {"Reserved",                                E_L0,  E_L3 ,  E_L0, E_L0},
@@ -459,7 +459,7 @@ static DEVICE_INFO D_APC0_Devices[] = {
     {"I2C0",                                    E_L0,  E_L3 ,  E_L0, E_L0},
     {"I2C1",                                    E_L0,  E_L3 ,  E_L0, E_L0},
     {"I2C2",                                    E_L0,  E_L3 ,  E_L0, E_L0},
-    {"SPI0",                                    E_L0,  E_L0 ,  E_L0, E_L0},
+    {"SPI0",                                    E_L1,  E_L3 ,  E_L0, E_L0},
     {"PTP_THERMAL_CTL",                         E_L0,  E_L0 ,  E_L0, E_L0},
     {"BTIF",                                    E_L0,  E_L3 ,  E_L0, E_L0},
     {"Reserved",                                E_L0,  E_L3 ,  E_L0, E_L0},
diff --git a/mediatek/proprietary/trustzone/custom/build/project/nd387b.mk b/mediatek/proprietary/trustzone/custom/build/project/nd387b.mk
index 17d688d..3d6f4e8 100755
--- a/mediatek/proprietary/trustzone/custom/build/project/nd387b.mk
+++ b/mediatek/proprietary/trustzone/custom/build/project/nd387b.mk
@@ -1,7 +1,7 @@
 MTK_ATF_SUPPORT = yes
-MTK_TEE_SUPPORT = no
+MTK_TEE_SUPPORT = yes
 TRUSTONIC_TEE_SUPPORT = no
 MTK_MACH_TYPE=mt6735m
-MICROTRUST_TEE_SUPPORT = no
+MICROTRUST_TEE_SUPPORT = yes
 MTK_SOTER_SUPPORT = no
 MTK_TEE_DRAM_SIZE = 0x3c00000
