diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 4c2b57e..c3989f2 100755
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -3465,6 +3465,12 @@ for nck code
             </intent-filter>
         </receiver>
 
+        <receiver android:name="com.android.settings.SimStateChangeReceiver" >
+            <intent-filter>
+                <action android:name="android.intent.action.SIM_STATE_CHANGED" />
+            </intent-filter>
+        </receiver>
+
         <activity
             android:name="com.v.sos.ContactsSettings"
             android:finishOnTaskLaunch="true"
diff --git a/src/com/android/settings/SimStateChangeReceiver.java b/src/com/android/settings/SimStateChangeReceiver.java
new file mode 100644
index 0000000..b333f7c
--- /dev/null
+++ b/src/com/android/settings/SimStateChangeReceiver.java
@@ -0,0 +1,34 @@
+package com.android.settings;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.SystemProperties;
+import android.util.Log;
+import android.content.pm.PackageManager;
+
+
+public class SimStateChangeReceiver extends BroadcastReceiver {
+
+    private final static String TAG = "SimStateChangeReceiver";
+
+    private String APTG_MCC_MNC = "46605";//APTG operator mcc mnc info 46605
+    @Override
+        public void onReceive(Context context, Intent intent) {
+            Log.d(TAG, "onReceive");
+            String mccMnc = "null";
+            mccMnc = SystemProperties.get("gsm.sim.operator.numeric", "0");
+            boolean isShowApp = mccMnc.contains(APTG_MCC_MNC);
+            PackageManager pm = context.getPackageManager();
+            try {
+                if (isShowApp) {
+                    pm.setApplicationEnabledSetting("com.aptg.gtapp", PackageManager.COMPONENT_ENABLED_STATE_DEFAULT, 0);
+                } else {
+                    pm.setApplicationEnabledSetting("com.aptg.gtapp", PackageManager.COMPONENT_ENABLED_STATE_DISABLED_USER, 0);
+                }
+            } catch (IllegalArgumentException exception) {
+                //do noting
+            }
+        }
+
+}
diff --git a/src/com/android/settings/applications/ManageApplications.java b/src/com/android/settings/applications/ManageApplications.java
index 16b9483..e45875c 100644
--- a/src/com/android/settings/applications/ManageApplications.java
+++ b/src/com/android/settings/applications/ManageApplications.java
@@ -1043,6 +1043,9 @@ public class ManageApplications extends InstrumentedFragment
                 if(hideApps.contains(nlabel)){
                     showEntries.remove(entry);
                 }
+                if (!entry.info.enabled && "com.aptg.gtapp".equals(nlabel)) {
+                    showEntries.remove(entry);
+                }
             }
            origEntries = showEntries;
 // End of Vanzo: zhongyeqing
