diff --git a/mediatek/mt6735/device.mk b/mediatek/mt6735/device.mk
old mode 100644
new mode 100755
index 84180dd..caa8896
--- a/mediatek/mt6735/device.mk
+++ b/mediatek/mt6735/device.mk
@@ -1374,7 +1374,9 @@ endif
 ifeq ($(strip $(MICROTRUST_TEE_SUPPORT)), yes)
   PRODUCT_COPY_FILES += device/mediatek/mt6735/init.microtrust.rc:root/init.microtrust.rc
   PRODUCT_COPY_FILES += vendor/mediatek/proprietary/trustzone/microtrust/source/platform/mt6735/teei/soter.raw:$(TARGET_COPY_OUT_VENDOR)/thh/soter.raw:mtk
-  PRODUCT_COPY_FILES += vendor/mediatek/proprietary/trustzone/microtrust/source/platform/mt6735/teei/fp_server:$(TARGET_COPY_OUT_VENDOR)/thh/fp_server:mtk
+  #PRODUCT_COPY_FILES += vendor/mediatek/proprietary/trustzone/microtrust/source/platform/mt6735/teei/fp_server:$(TARGET_COPY_OUT_VENDOR)/thh/fp_server:mtk
+  PRODUCT_COPY_FILES += vendor/mediatek/proprietary/trustzone/microtrust/source/platform/mt6735/teei/fp_server_elan:$(TARGET_COPY_OUT_VENDOR)/thh/fp_server_elan:mtk
+  PRODUCT_COPY_FILES += vendor/mediatek/proprietary/trustzone/microtrust/source/platform/mt6735/teei/fp_server_blestech:$(TARGET_COPY_OUT_VENDOR)/thh/fp_server_blestech:mtk
   PRODUCT_COPY_FILES += vendor/mediatek/proprietary/trustzone/microtrust/source/platform/mt6735/teei/init_thh:$(TARGET_COPY_OUT_VENDOR)/bin/init_thh:mtk
   PRODUCT_COPY_FILES += vendor/mediatek/proprietary/trustzone/microtrust/source/platform/mt6735/teei/uTAgent:$(TARGET_COPY_OUT_VENDOR)/thh/uTAgent:mtk
   PRODUCT_PACKAGES += keystore.mt6735
@@ -1618,7 +1620,8 @@ ifeq ($(strip $(VANZO_BLESTECH_TEE_FP32)), yes)
     PRODUCT_PACKAGES += fingerprintd
     PRODUCT_COPY_FILES +=vendor/mediatek/proprietary/external/blestech_tee/blestech.fingerprint.default.so:system/lib/hw/blestech.fingerprint.default.so
     PRODUCT_COPY_FILES +=vendor/mediatek/proprietary/external/blestech_tee/libBtlFpHal.so:system/lib/libBtlFpHal.so
-	PRODUCT_COPY_FILES +=vendor/mediatek/proprietary/external/blestech/libxuFPAlg.so:system/lib/libxuFPAlg.so
+   PRODUCT_COPY_FILES +=vendor/mediatek/proprietary/external/blestech_tee/libxuFPAlg.so:system/lib/libxuFPAlg.so
+	PRODUCT_COPY_FILES +=vendor/mediatek/proprietary/external/blestech_tee/libbtlfp.so:system/lib/libbtlfp.so
 endif
 
 # End of Vanzo: yangbinbin
@@ -1631,3 +1634,9 @@ ifeq ($(strip $(VANZO_FPSENSOR_64BIT_FINGERPRINT)), yes)
 	PRODUCT_COPY_FILES +=vendor/mediatek/proprietary/external/fpsensor/lib64/fingerprint.default.so:system/lib64/hw/fingerprint.default.so
 	PRODUCT_COPY_FILES +=vendor/mediatek/proprietary/external/fpsensor/lib64/fpsensor_module.default.so:system/lib64/hw/fpsensor_module.default.so
 endif
+
+
+#blestech add
+PRODUCT_PACKAGES += BtlFingerprint
+PRODUCT_PACKAGES += libbtlfp
+#blestech end
